src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -2079,6 +2099,120 @@ TEST_CASE("temp entry eviction", "[tx][soroban]")
+    {
+        auto tx = test.createRestoreTx(restoreResources, 1'000,
+                                       40'000 * contractKeys.size());
+        test.invokeTx(tx, false);
+        REQUIRE(tx->getResult()
+                    .result.results()[0]
+                    .tr()
+                    .restoreFootprintResult()
+                    .code() == RESTORE_FOOTPRINT_INSUFFICIENT_REFUNDABLE_FEE);
+    }
+
+    // exceeded readBytes
+    {
+        auto resourceCopy = restoreResources;
+        resourceCopy.readBytes = 1;
+        auto tx = test.createRestoreTx(resourceCopy, 1'000, resourceFee);
+        test.invokeTx(tx, false);
+        REQUIRE(tx->getResult()
+                    .result.results()[0]
+                    .tr()
+                    .restoreFootprintResult()
+                    .code() == RESTORE_FOOTPRINT_RESOURCE_LIMIT_EXCEEDED);
+    }
+
+    // exceeded writeBytes
+    {
+        auto resourceCopy = restoreResources;
+        resourceCopy.writeBytes = 1;
+        auto tx = test.createRestoreTx(resourceCopy, 1'000, resourceFee);
+        test.invokeTx(tx, false);
+        REQUIRE(tx->getResult()
+                    .result.results()[0]
+                    .tr()
+                    .restoreFootprintResult()
+                    .code() == RESTORE_FOOTPRINT_RESOURCE_LIMIT_EXCEEDED);
+    }
+
+    {
+        // restore entries
+        auto tx = test.createRestoreTx(restoreResources, 1'000, resourceFee);
+        test.invokeTx(tx, true);
+    }
+
+    ledgerSeq = test.getLedgerSeq();
+    REQUIRE(test.isEntryLive(contractKeys[0], ledgerSeq));
+    REQUIRE(test.isEntryLive(contractKeys[1], ledgerSeq));
+
+    SorobanResources extendResources;
+    extendResources.footprint.readOnly = contractKeys;
+    extendResources.instructions = 0;
+    extendResources.readBytes = 10'000;
+    extendResources.writeBytes = 0;
+
+    // exceeded readBytes
+    {
+        auto resourceCopy = extendResources;
+        resourceCopy.readBytes = 10;
+
+        auto resourceFee = DEFAULT_TEST_RESOURCE_FEE * contractKeys.size();
+        auto tx =
+            test.createExtendOpTx(resourceCopy, 10'000, 1'000, resourceFee);
+        test.invokeTx(tx, false);
+        REQUIRE(tx->getResult()
+                    .result.results()[0]
+                    .tr()
+                    .extendFootprintTTLResult()
+                    .code() == EXTEND_FOOTPRINT_TTL_RESOURCE_LIMIT_EXCEEDED);
+    }
+
+    // insufficient refundable fee
+    {
+        auto resourceCopy = extendResources;
+
+        auto resourceFee = DEFAULT_TEST_RESOURCE_FEE * contractKeys.size();
+        auto tx =
+            test.createExtendOpTx(extendResources, 10'000, 30'000, 30'000);
+        test.invokeTx(tx, false);
+        REQUIRE(tx->getResult()
+                    .result.results()[0]
+                    .tr()
+                    .extendFootprintTTLResult()
+                    .code() ==
+                EXTEND_FOOTPRINT_TTL_INSUFFICIENT_REFUNDABLE_FEE);
```

Comment:
Can we also add checks for `EXTEND_FOOTPRINT_TTL_MALFORMED` and `RESTORE_FOOTPRINT_MALFORMED`? I think these probably belong in the `invalid footprint keys` tests. We already check that the TX is not valid, but we don't check the code. We should probably also add a check for `INVOKE_HOST_FUNCTION_MALFORMED` there as well.

Reply:
Done